src/java.base/share/classes/java/io/BufferedReader.java

```diff
@@ -394,12 +395,48 @@ String readLine(boolean ignoreLF, boolean[] term) throws IOException {
      *
      * @throws     IOException  If an I/O error occurs
      *
+     * @see #readAllLines
      * @see java.nio.file.Files#readAllLines
      */
     public String readLine() throws IOException {
         return readLine(false, null);
     }
 
+    /**
+     * Reads all remaining lines of text. A line is considered to be terminated
+     * by any one of a line feed ('\n'), a carriage return ('\r'), a carriage
+     * return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
+     * <p> This method works as if invoking it were equivalent to evaluating
+     * the expression:
+     * <blockquote>{@linkplain #lines()}.toList()</blockquote>
+     * The method does not close this reader nor its underlying stream.
+     * If an I/O error occurs, the states of the reader and its underlying
+     * stream are unspecified.
+     *
+     * @apiNote
+     * This method is intended for simple cases where it is convenient
+     * to read all remaining lines in a single operation. It is not intended for
+     * reading a large number of lines.
+     *
+     * @return     the remaining lines of text as an unmodifiable {@code List}
+     *
+     * @throws     IOException  If an I/O error occurs
+     *
+     * @see java.nio.file.Files#readAllLines
+     *
+     * @since 25
+     */
+    public List<String> readAllLines() throws IOException {
+        try {
+            return lines().toList();
```

Comment:
I'd suggest not having this method's implementation depend on an overridable method.

Reply:
> I'd suggest not having this method's implementation depend on an overridable method.

I'm jumping the gun here by referring to an implementation detail, but the overridable method problem could be circumvented using an approach taken in other classes in `java.io`:
```
        if (getClass() == BufferedReader.class) {
            try {
                return lines().toList();
            } catch (UncheckedIOException ue) {
                // the cause cannot be null
                throw ue.getCause();
            }
        } else {
            return super.readAllLines();
        }
```
This would not need to be mentioned in the specification.